### DISCUSSION TITLE
How to connect my EspMQTTClient object to tagocore using MQTT

### DISCUSSION THREAD

**Discussion by yessicavalencia44** [tago-io/tcore#43] [Q&A] [answered]
Hi, I want to connect my ESP32 board to tagocore using "EspMQTTClient.h" library. This library uses an object called EspMQTTClient client. To connect with Tagoio in the cloud, it is done as follows.

**EspMQTTClient client
(
  "Wi-Fi network name", 
  "password for your Wi-Fi network", 
  "mqtt.tago.io",  // MQTT Broker server ip 
  "Default",   // username
  "Token for device",   // Token
  "TestClient",     // Client name that uniquely identify your device
  1883              // The MQTT port, default to 1883. this line can be omitted
);**

In the case of Tagocore, I am asking for help to have a connection. Thanks

**Comment by RicardoStoklosa** [comment, marked as answer]
Hi @yessicavalencia44!

The TagoCore MQTT plugin works very similar to TagoIO, your example should look like this:

EspMQTTClient client
(
"Wi-Fi network name",
"password for your Wi-Fi network",
"localhost", // Localhost or your tagocore ip
"token", // To work with device token
"device token", // Your tagoCore Device Token
"TestClient", // Client name that uniquely identify your device
1883 // The MQTT port, default to 1883. this line can be omitted
);

More information it's available in the plugin docs.
